151 implement volunteer side home page

theNatePi opened the pull request:
## Changes
<details><summary>Volunteer Home Page</summary>
<p>

- Copied admin functionality onto volunteer-home route
- Removed selection and archiving functionality
- Maintained Search Location and Search Date functionality although it was not in design (I believe we talked about keeping this included before)
- Slightly change EventCard to match design on Figma
- Implement hamburger menu

<img width="1229" alt="image" src="https://github.com/ctc-uci/stand-up-to-trash-frontend/assets/78774649/4f1c29ae-92cc-448d-b812-cef3537ed810">

<img width="467" alt="image" src="https://github.com/ctc-uci/stand-up-to-trash-frontend/assets/78774649/d3b98737-e6b7-47ec-a150-73bef917badb">

</p>
</details> 

<details><summary>Impact Summary</summary>
<p>

- Implemented Impact Summary and proper icons
- Updated DataCard to match design

<img width="1229" alt="image" src="https://github.com/ctc-uci/stand-up-to-trash-frontend/assets/78774649/6e0fe974-4347-45ae-b8c3-a070f62b2a99">
</p>
</details> 

## Testing
- Added line in VolunteerImpactSummary.jsx to set user ID to 111, this user has available events for testing values
    - <img width="375" alt="image" src="https://github.com/ctc-uci/stand-up-to-trash-frontend/assets/78774649/b46b3d36-5bbb-4032-af37-35968e57445f">
- Implemented check for isNaN(largestItem) || !isFinite(largestItem) if provided weights are empty or invalid

gayathriy22 committed:
unfinished impact summary (mock)

theNatePi committed:
Implement functionality for ImpactSummary

theNatePi committed:
Add comments, remove console logs

theNatePi committed:
Remove admin archive functionality

theNatePi committed:
Add icons file for Impact Summary

theNatePi committed:
Match design

theNatePi committed:
Update to match design

theNatePi committed:
Match design better

theNatePi committed:
Replace testing ID with prod

ctc-devops commented on src/components/Events/VolunteerImpactSummary.jsx:
```diff
@@ -0,0 +1,112 @@
+    updateUser();
+    console.log(user);
+    getData();
+  }, []);
```
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'getData', 'updateUser', and 'user'. Either include them or remove the dependency array.

ctc-devops commented on src/pages/VolunteerHomePage.jsx:
```diff
@@ -0,0 +1,176 @@
+      } else result = events;
+      console.log(result);
+      setDisplayEvents(result);
+    }, [name, location, date, fuse]);
```
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'events'. Either include it or remove the dependency array.

theNatePi committed:
Add lbs label to trash weight cards

I forgot to do this initially, match design by adding 'lbs' next to the Total Trash Weight and Largest Trash Item Collected

NwinNwin reviewed (approved):
LGTM Tech lead

NwinNwin committed:
merged main
